frontend/lib/resolution/Resolver.cpp

```diff
@@ -2997,14 +3020,18 @@ void Resolver::exit(const Dot* dot) {
                           LOOKUP_IMPORT_AND_USE |
                           LOOKUP_EXTERN_BLOCKS;
 
+    if (!moduleId.contains(dot->id())) {
+      config |= LOOKUP_SKIP_PRIVATE_VIS;
+    }
```

Comment:
This makes me think of issue #21723 -- not sure if moduleId.contains is the right move for some interpretations. That said, matching production right now is a good idea.